plugins/modules/accesspoint_workflow_manager.py

```diff
@@ -0,0 +1,818 @@
+        self.payload = module.params
+        self.keymap = {}
+
+    # Below function used to validate input over the ansible validation
+    def validate_input_yml(self):
+        """
+        Validate the fields provided in the yml files.
+        Checks the configuration provided in the playbook against a predefined specification
+        to ensure it adheres to the expected structure and data types based on input.
+        Parameters:
+          - self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Returns:
+          The method returns an instance of the class with updated attributes:
+                - self.msg: A message describing the validation result.
+                - self.status: The status of the validation (either 'success' or 'failed').
+                - self.validated_config: If successful, a validated version of the 'config' parameter.
+        Example:
+            To use this method, create an instance of the class and call 'validate_input_yml' on it.
+            If the validation succeeds, 'self.status' will be 'success' and 'self.validated_config'
+            will contain the validated configuration. If it fails, 'self.status' will be 'failed', and
+            'self.msg' will describe the validation issues.To use this method, create an instance of the class and call 'validate_input_yml' on it.
+          If the validation succeeds, this will allow to go next step, unless this will stop execution.
+          based on the fields.
+        """
+        self.log('Validating the Playbook Yaml File..', "INFO")
+        try:
+            errormsg = []
+            aplist = self.payload.get("config")
+            aplist = self.camel_to_snake_case(aplist)
+            aplist = self.update_site_type_key(aplist)
+            accesspoint_spec = dict(mac_address=dict(required=False, type='str'),
+                        led_brightness_level=dict(required=False, type='int'),
+                        led_status = dict(required=False, type='str'),
+                        location = dict(required=False, type='str'),
+                        ap_name = dict(required=False, type='str'),
+                        ap_name_new = dict(required=False, type='str'),
+                        management_ip_address = dict(required=False, type='str'),
+                        hostname = dict(required=False, type='str'),
+                        device_fields = dict(required=False, type='str'),
+                        ap_selected_field = dict(required=False, type='str'),
+                        )
+            valid_param, invalid_param = validate_list_of_dicts(aplist, accesspoint_spec)
+            eachap = valid_param[0]
+            if len(invalid_param) > 0:
+                errormsg.append("Invalid param found in playbook: '{0}' "\
+                                .format(", ".join(invalid_param)))
+            self.log(str(eachap) + str(valid_param), "INFO")
+            if eachap.get("mac_address"):
```

Comment:
These validation should be added in the get_diff_<state> where it used/required. As we are checking everything irrespective of operation so it will slow the execution process.

Reply:
moved specific field inside get_diff_merged